### PR TITLE
Task 3: Abilities, Inventory and Ui refactoring

### PR DESCRIPTION
**Pull request by gabrielcicero45**
[Gravação de tela de 02-02-2023 13:46:51.webm](https://user-images.githubusercontent.com/42891342/216388204-8bf7abc2-427c-4cf9-bae0-eaade6acf525.webm)

### CODE REVIEW

File: src/components/Attributes/Attributes.js [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
-    <>
-      <div className="sheet__item">Strength: {strength}</div>
-      <div className="sheet__item">Dexterity: {dexterity}</div>
-      <div className="sheet__item">Agility: {agility}</div>
-      <div className="sheet__item">Intelligency: {intelligency}</div>
-      <div className="sheet__item">Wisdom: {wisdom}</div>
-      <div className="sheet__item">Charisma: {charisma}</div>
-    </>
-  )
+    <table className="sheet__item">
```
Comment (julialbq) [inline src/components/Attributes/Attributes.js]: Try not to use the tag table for style, just when you are rendering data as a table

File: src/components/Ability/Ability.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+export default function Ability({name,description,manaCost}) {
+  return (
+    <tr>
```
Comment (julialbq) [inline src/components/Ability/Ability.js:5]: Try not to use the tag table for style, just when you are rendering data as a table

File: src/components/Inventory/Inventory.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import InventoryItem from "../InventoryItem/InventoryItem";
+export default function Inventory({ inventory, strength }) {
+  let totalWeight = 0;
+  inventory.forEach((item) => {
+    totalWeight += Object.values(item)[0].weight * item.amount;
+  });
+  return (
+    <table>
```
Comment (julialbq) [inline src/components/Inventory/Inventory.js:9]: Try not to use the tag table for style, just when you are rendering data as a table

File: src/components/InventoryItem/InventoryItem.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+export default function InventoryItem({item}) {
+  const name = Object.keys(item)[0].replace('_', ' ')
+  const {description,weight,type} = Object.values(item)[0];
+  return (
+    <tr>
```
Comment (julialbq) [inline src/components/InventoryItem/InventoryItem.js:7]: Try not to use the tag table for style, just when you are rendering data as a table

**Review by julialbq — APPROVED** [review]

File: src/components/Attributes/Attributes.js [hunk 5]
```diff
@@ -1,15 +1,35 @@
-import React from 'react'
+import React from "react";
 
-export default function Attributes({attributes}) {
-  const {strength,dexterity,agility,intelligency,wisdom,charisma} = attributes
+export default function Attributes({ attributes }) {
+  const { strength, dexterity, agility, intelligency, wisdom, charisma } =
+    attributes;
```
Comment (negreirosleo) [inline src/components/Attributes/Attributes.js]:
```suggestion
export default function Attributes({ strength, dexterity, agility, intelligency, wisdom, charisma }) {
```

Comment (negreirosleo) [reply src/components/Attributes/Attributes.js]: Outra forma de fazer isso de maneira mais programática
```js
Object.entries(attributes).map(
  ([name, value]) => <td className="sheet__item">{name}: {value}</td>
)
```

Dessa forma transformamos nosso objeto em um array com o nome e o valor do attributo e renderizamos com apenas um map!

File: src/components/Inventory/Inventory.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import InventoryItem from "../InventoryItem/InventoryItem";
+export default function Inventory({ inventory, strength }) {
+  let totalWeight = 0;
+  inventory.forEach((item) => {
+    totalWeight += Object.values(item)[0].weight * item.amount;
+  });
```
Comment (negreirosleo) [inline src/components/Inventory/Inventory.js]: textbook reduce:
```js
  const totalWeight = inventory.reduce((accumulator, currentValue) => {
    return accumulator + Object.values(currentValue)[0].weight * currentValue.amount;
  }, 0);
  ```

File: src/components/Inventory/Inventory.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        <tr>
+          <td colSpan={2}></td>
+          <td>Total Weight: </td>
+          <td colSpan={2}>{totalWeight.toFixed(2)}</td>
+        </tr>
+        <tr>
+          <td colSpan={2}></td>
+          <td>Weight Status:</td>
+          <td colSpan={2}>
+            {totalWeight > strength * 2 ? "Overweight" : "Ok"}
```
Comment (negreirosleo) [inline src/components/Inventory/Inventory.js]: Seria legal ter esse strength * 2 numa variavel

File: src/components/Sheet/Sheet.js [hunk 8]
```diff
@@ -1,52 +1,48 @@
...
+  
   return (
     <>
-    <div className='sheet__container'>
-      <div className="sheet__column">
-        <Name playerName={name}/>
+      <div className="sheet__container">
+        <div className="sheet__column">
+          <RpgClass rpgClass={rpgClass} />
+          <Name name={name} surname={surmane} />
```
Comment (gabrielcicero45) [inline src/components/Sheet/Sheet.js]:
```suggestion
          <Name name={name} surname={surname} />
```

File: src/components/Sheet/Sheet.js [hunk 9]
```diff
@@ -1,52 +1,48 @@
...
+import RpgClass from "../RpgClass/RpgClass";
+import HealthAndMana from "../HealthAndMana/HealthAndMana";
+import LevelAndXp from "../LevelAndXp/LevelAndXp";
+import Name from "../Name/Name";
+import "./Sheet.css";
 
+import Inventory from "../Inventory/Inventory";
+import AbilitiesList from "../AbilitiesList/AbilitiesList";
+export default function Sheet({ player }) {
+  const { name, surmane, health, mana, level, xp, attributes, rpgClass, abilities, inventory } =
```
Comment (gabrielcicero45) [inline src/components/Sheet/Sheet.js]:
```suggestion
  const { name, surname, health, mana, level, xp, attributes, rpgClass, abilities, inventory } =
```

**Pull request merged**